Helm charts improvement

Hi there,

We are planning to deploy the application into Production soon, but we have a pre-req that the application must use sealed secrets instead of declaring the credentials into the values.yaml files.

Would it be possible to:
1 - modify the charts to use secrets and manage those using sealed secrets charts?
2 - create a unique helm charts wrapping the all other helm charts so it can be stored into artifactory as a unique component
3 - authenticate the connection with CRDB using der and pk8 file instead of login/password as the example jdbc:postgresql://cockroachdb.amazonaws.com:26257/DATABASENAME?user=USERNAME&ApplicationName=APPNAME&sslcert=%2Fcerts%2Fpt1%2FFILENAME.der&sslkey=%2Fcerts%2Fpt1%2FFILENAME.pk8&sslmode=require&ssl=true

Kind regards,
Fabio

Hi @admin-vizcentric,

When I execute "helm lint <module name>" it returns the error below:
Could you please check it?

Kind regards,
Fabio

**helm lint flair-bi**
==> Linting flair-bi
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "flair-bi/templates/deployment.yaml": template: flair-bi/templates/deployment.yaml:37:14: executing "flair-bi/templates/deployment.yaml" at <.Values.secretEnv.ke...>: can't evaluate field keys in type interface {}

Error: 1 chart(s) linted, 1 chart(s) failed

**helm lint flair-cache**
==> Linting flair-cache
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "flair-cache/templates/deployment.yaml": template: flair-cache/templates/deployment.yaml:37:14: executing "flair-cache/templates/deployment.yaml" at <.Values.secretEnv.ke...>: can't evaluate field keys in type interface {}

Error: 1 chart(s) linted, 1 chart(s) failed

**helm lint flair-couchdb**                                                                                                                                                                                               1 ↵
==> Skipping flair-couchdb
No chart found for linting (missing Chart.yaml)

Error: 0 chart(s) linted, 1 chart(s) failed

**helm lint flair-engine**                                                                                                                                                                                                1 ↵
==> Linting flair-engine
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "flair-engine/templates/deployment.yaml": template: flair-engine/templates/deployment.yaml:37:14: executing "flair-engine/templates/deployment.yaml" at <.Values.secretEnv.ke...>: can't evaluate field keys in type interface {}

Error: 1 chart(s) linted, 1 chart(s) failed

**helm lint flair-notifications**                                                                                                                                                                                         1 ↵
==> Linting flair-notifications
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "flair-notifications/templates/deployment.yaml": template: flair-notifications/templates/deployment.yaml:38:14: executing "flair-notifications/templates/deployment.yaml" at <.Values.secretEnv.ke...>: can't evaluate field keys in type interface {}

Error: 1 chart(s) linted, 1 chart(s) failed

**helm lint flair-registry**                                                                                                                                                                                              1 ↵
==> Linting flair-registry
[INFO] Chart.yaml: icon is recommended
[ERROR] templates/: render error in "flair-registry/templates/deployment.yaml": template: flair-registry/templates/deployment.yaml:37:14: executing "flair-registry/templates/deployment.yaml" at <.Values.secretEnv.ke...>: can't evaluate field keys in type interface {}

Error: 1 chart(s) linted, 1 chart(s) failed

--- a/README.md
+++ b/README.md
@@ -47,7 +47,7 @@ Pull down all necessary images for deployment
 docker pull flairbi/flair-registry:v5.0.6.ae1e037
 docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
-docker pull flairbi/flair-notifications:2.5.1
+docker pull flairbi/flair-notifications:2.5.2
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
 docker pull flairbi/flairbi:v2.5.1

--- a/README.md
+++ b/README.md
@@ -45,11 +45,11 @@ Pull down all necessary images for deployment
 
 ```sh
 docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:latest
-docker pull flairbi/flair-cache:latest
-docker pull flairbi/flair-notifications:latest
+docker pull flairbi/flair-engine:v2.2.5-SNAPSHOT.c873a96
+docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
+docker pull flairbi/flair-notifications:2.1.3-SNAPSHOT.119
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:latest
+docker pull flairbi/flairbi:v2.2.1-SNAPSHOT.e1a39e43
 ```
 
 ### Repository and environment setup

--- a/README.md
+++ b/README.md
@@ -45,11 +45,11 @@ Pull down all necessary images for deployment
 
 ```sh
 docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:v2.2.5-SNAPSHOT.c873a96
+docker pull flairbi/flair-engine:v2.4.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
-docker pull flairbi/flair-notifications:2.1.3-SNAPSHOT.119
+docker pull flairbi/flair-notifications:2.4.1
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:v2.2.1-SNAPSHOT.e1a39e43
+docker pull flairbi/flairbi:v2.4.1-SNAPSHOT.eb12f144
 ```
 
 ### Repository and environment setup

--- a/README.md
+++ b/README.md
@@ -44,7 +44,7 @@ Since we now have the prerequisites necessary for us to deploy flair services so
 Pull down all necessary images for deployment
 
 ```sh
-docker pull flairbi/flair-registry:v5.0.8
+docker pull flairbi/flair-registry:v5.0.6.ae1e037
 docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
 docker pull flairbi/flair-notifications:2.5.1

--- a/README.md
+++ b/README.md
@@ -44,12 +44,13 @@ Since we now have the prerequisites necessary for us to deploy flair services so
 Pull down all necessary images for deployment
 
 ```sh
-docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:v2.4.0
+docker pull flairbi/flair-registry:v5.0.8
+docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
-docker pull flairbi/flair-notifications:2.4.1
+docker pull flairbi/flair-notifications:2.5.1
+docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:v2.4.1-SNAPSHOT.eb12f144
+docker pull flairbi/flairbi:v2.5.1
 ```
 
 ### Repository and environment setup

--- a/flair-bi/templates/deployment.yaml
+++ b/flair-bi/templates/deployment.yaml
@@ -25,11 +25,22 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-{{- if .Values.env }}
           env:
+            - name: FLAIR_COMPONENT
+              value: "flair-bi"
+{{- if .Values.env }}
     {{- range $key, $val := .Values.env }}
             - name: {{ $key }}
               value: {{ $val | quote }}
+    {{- end}}
+{{- end }}
+{{- if .Values.secretEnv.keys }}
+    {{- range .Values.secretEnv.keys }}
+            - name: {{ . }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Values.secretEnv.name }}
+                  key: {{ . }}
     {{- end}}
 {{- end }}
           ports:
@@ -50,7 +61,7 @@ spec:
             timeoutSeconds: 5
           resources:
 {{ toYaml .Values.resources | indent 12 }}
-        {{- if .Values.secretName }}
+        {{- if .Values.secretMountName }}
           volumeMounts:
             - name: secrets
               mountPath: "/opt/flair/secrets"
@@ -68,9 +79,9 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
     {{- end }}
-    {{- if .Values.secretName }}
+    {{- if .Values.secretMountName }}
       volumes:
         - name: secrets
           secret:
-            secretName: {{ .Values.secretName }}
+            secretName: {{ .Values.secretMountName }}
     {{- end }}

--- a/flair-bi/templates/deployment.yaml
+++ b/flair-bi/templates/deployment.yaml
@@ -50,6 +50,12 @@ spec:
             timeoutSeconds: 5
           resources:
 {{ toYaml .Values.resources | indent 12 }}
+        {{- if .Values.secretName }}
+          volumeMounts:
+            - name: secrets
+              mountPath: "/opt/flair/secrets"
+              readOnly: true
+        {{- end }}
     {{- with .Values.nodeSelector }}
       nodeSelector:
 {{ toYaml . | indent 8 }}
@@ -61,4 +67,10 @@ spec:
     {{- with .Values.tolerations }}
       tolerations:
 {{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- if .Values.secretName }}
+      volumes:
+        - name: secrets
+          secret:
+            secretName: {{ .Values.secretName }}
     {{- end }}

--- a/flair-bi/values.yaml
+++ b/flair-bi/values.yaml
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: flairbi/flairbi
-  tag: v2.2.1-SNAPSHOT.e1a39e43
+  tag: v2.5.1
   pullPolicy: Always
 
 nameOverride: ""

--- a/flair-bi/values.yaml
+++ b/flair-bi/values.yaml
@@ -6,11 +6,13 @@ replicaCount: 1
 
 image:
   repository: flairbi/flairbi
-  tag: latest
+  tag: v2.2.1-SNAPSHOT.e1a39e43
   pullPolicy: Always
 
 nameOverride: ""
 fullnameOverride: ""
+
+# secretName: flair-bi-secret
 
 service:
   type: ClusterIP

--- a/flair-bi/values.yaml
+++ b/flair-bi/values.yaml
@@ -12,7 +12,8 @@ image:
 nameOverride: ""
 fullnameOverride: ""
 
-# secretName: flair-bi-secret
+# Secret mount to be used if any secret files are needed to be added to the container during startup
+# secretMountName: flair-secret
 
 service:
   type: ClusterIP
@@ -25,6 +26,16 @@ env:
   EUREKA_URL: http://admin:admin@flair-registry:8761/eureka
   POSTGRES_URI: flair-bi-pg-postgresql:5432
   GRPC_SSL_ENABLED: false
+
+# Can be used if you want to set environment variables from a secret
+# The key names need to be the same as the environment variables you want to set
+# You would have to make sure the secret is created with the same key names as the environment variables you would like to set
+
+# secretEnv:
+#   name: flair-secret
+#   keys:
+#     - FLAIR_SECRET_USERNAME
+#     - FLAIR_SECRET_PASSWORD
 
 resources: {}
 

--- a/flair-cache/templates/deployment.yaml
+++ b/flair-cache/templates/deployment.yaml
@@ -53,6 +53,12 @@ spec:
             timeoutSeconds: 5
           resources:
 {{ toYaml .Values.resources | indent 12 }}
+        {{- if .Values.secretName }}
+          volumeMounts:
+            - name: secrets
+              mountPath: "/opt/flair/secrets"
+              readOnly: true
+        {{- end }}
     {{- with .Values.nodeSelector }}
       nodeSelector:
 {{ toYaml . | indent 8 }}
@@ -64,4 +70,10 @@ spec:
     {{- with .Values.tolerations }}
       tolerations:
 {{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- if .Values.secretName }}
+      volumes:
+        - name: secrets
+          secret:
+            secretName: {{ .Values.secretName }}
     {{- end }}

--- a/flair-cache/templates/deployment.yaml
+++ b/flair-cache/templates/deployment.yaml
@@ -25,11 +25,22 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-{{- if .Values.env }}
           env:
+            - name: FLAIR_COMPONENT
+              value: "flair-cache"
+{{- if .Values.env }}
     {{- range $key, $val := .Values.env }}
             - name: {{ $key }}
               value: {{ $val | quote }}
+    {{- end}}
+{{- end }}
+{{- if .Values.secretEnv.keys }}
+    {{- range .Values.secretEnv.keys }}
+            - name: {{ . }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Values.secretEnv.name }}
+                  key: {{ . }}
     {{- end}}
 {{- end }}
           ports:
@@ -53,7 +64,7 @@ spec:
             timeoutSeconds: 5
           resources:
 {{ toYaml .Values.resources | indent 12 }}
-        {{- if .Values.secretName }}
+        {{- if .Values.secretMountName }}
           volumeMounts:
             - name: secrets
               mountPath: "/opt/flair/secrets"
@@ -71,9 +82,9 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
     {{- end }}
-    {{- if .Values.secretName }}
+    {{- if .Values.secretMountName }}
       volumes:
         - name: secrets
           secret:
-            secretName: {{ .Values.secretName }}
+            secretName: {{ .Values.secretMountName }}
     {{- end }}

--- a/flair-cache/values.yaml
+++ b/flair-cache/values.yaml
@@ -6,11 +6,13 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-cache
-  tag: latest
+  tag: v2.2.2-SNAPSHOT.2f59747
   pullPolicy: Always
 
 nameOverride: ""
 fullnameOverride: ""
+
+# secretName: flair-cache-secret
 
 service:
   type: ClusterIP

--- a/flair-cache/values.yaml
+++ b/flair-cache/values.yaml
@@ -12,7 +12,8 @@ image:
 nameOverride: ""
 fullnameOverride: ""
 
-# secretName: flair-cache-secret
+# Secret mount to be used if any secret files are needed to be added to the container during startup
+# secretMountName: flair-secret
 
 service:
   type: ClusterIP
@@ -24,6 +25,16 @@ env:
   CONFIG_URI: http://flair-registry:8761
   EUREKA_URL: http://admin:admin@flair-registry:8761/eureka
   GRPC_SSL_ENABLED: false
+
+# Can be used if you want to set environment variables from a secret
+# The key names need to be the same as the environment variables you want to set
+# You would have to make sure the secret is created with the same key names as the environment variables you would like to set
+
+# secretEnv:
+#   name: flair-secret
+#   keys:
+#     - FLAIR_SECRET_USERNAME
+#     - FLAIR_SECRET_PASSWORD
 
 resources: {}
 

--- a/flair-engine/templates/deployment.yaml
+++ b/flair-engine/templates/deployment.yaml
@@ -25,11 +25,22 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-{{- if .Values.env }}
           env:
+            - name: FLAIR_COMPONENT
+              value: "flair-engine"
+{{- if .Values.env }}
     {{- range $key, $val := .Values.env }}
             - name: {{ $key }}
               value: {{ $val | quote }}
+    {{- end}}
+{{- end }}
+{{- if .Values.secretEnv.keys }}
+    {{- range .Values.secretEnv.keys }}
+            - name: {{ . }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Values.secretEnv.name }}
+                  key: {{ . }}
     {{- end}}
 {{- end }}
           ports:
@@ -57,7 +68,7 @@ spec:
             timeoutSeconds: 5
           resources:
 {{ toYaml .Values.resources | indent 12 }}
-        {{- if .Values.secretName }}
+        {{- if .Values.secretMountName }}
           volumeMounts:
             - name: secrets
               mountPath: "/opt/flair/secrets"
@@ -75,9 +86,9 @@ spec:
       tolerations:
 {{ toYaml . | indent 8 }}
     {{- end }}
-    {{- if .Values.secretName }}
+    {{- if .Values.secretMountName }}
       volumes:
         - name: secrets
           secret:
-            secretName: {{ .Values.secretName }}
+            secretName: {{ .Values.secretMountName }}
     {{- end }}

--- a/flair-engine/templates/deployment.yaml
+++ b/flair-engine/templates/deployment.yaml
@@ -57,6 +57,12 @@ spec:
             timeoutSeconds: 5
           resources:
 {{ toYaml .Values.resources | indent 12 }}
+        {{- if .Values.secretName }}
+          volumeMounts:
+            - name: secrets
+              mountPath: "/opt/flair/secrets"
+              readOnly: true
+        {{- end }}
     {{- with .Values.nodeSelector }}
       nodeSelector:
 {{ toYaml . | indent 8 }}
@@ -68,4 +74,10 @@ spec:
     {{- with .Values.tolerations }}
       tolerations:
 {{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- if .Values.secretName }}
+      volumes:
+        - name: secrets
+          secret:
+            secretName: {{ .Values.secretName }}
     {{- end }}

--- a/flair-engine/values.yaml
+++ b/flair-engine/values.yaml
@@ -12,7 +12,8 @@ image:
 nameOverride: ""
 fullnameOverride: ""
 
-# secretName: flair-engine-secret
+# Secret mount to be used if any secret files are needed to be added to the container during startup
+# secretMountName: flair-secret
 
 service:
   type: ClusterIP
@@ -25,6 +26,16 @@ env:
   EUREKA_URL: http://admin:admin@flair-registry:8761/eureka
   POSTGRES_URI: flair-engine-pg-postgresql:5432
   GRPC_SSL_ENABLED: false
+
+# Can be used if you want to set environment variables from a secret
+# The key names need to be the same as the environment variables you want to set
+# You would have to make sure the secret is created with the same key names as the environment variables you would like to set
+
+# secretEnv:
+#   name: flair-secret
+#   keys:
+#     - FLAIR_SECRET_USERNAME
+#     - FLAIR_SECRET_PASSWORD
 
 resources: {}
 

--- a/flair-engine/values.yaml
+++ b/flair-engine/values.yaml
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-engine
-  tag: v2.2.5-SNAPSHOT.c873a96
+  tag: v2.5.0
   pullPolicy: Always
 
 nameOverride: ""

--- a/flair-engine/values.yaml
+++ b/flair-engine/values.yaml
@@ -6,11 +6,13 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-engine
-  tag: latest
+  tag: v2.2.5-SNAPSHOT.c873a96
   pullPolicy: Always
 
 nameOverride: ""
 fullnameOverride: ""
+
+# secretName: flair-engine-secret
 
 service:
   type: ClusterIP

--- a/flair-notifications/templates/configmap.yaml
+++ b/flair-notifications/templates/configmap.yaml
@@ -10,7 +10,6 @@ data:
     grpcPort: 8091
     imageFolder: /flair-notifications/images
     compressImageFolder: /flair-notifications/compressImages
-    grpcEndPoint: flair-engine:6565
     discovery:
         eureka:
             url: http://admin:admin@flair-registry:8761/eureka

--- a/flair-notifications/templates/configmap.yaml
+++ b/flair-notifications/templates/configmap.yaml
@@ -9,8 +9,8 @@ data:
     httpPort: 8090
     grpcPort: 8091
     imageFolder: /flair-notifications/images
+    compressImageFolder: /flair-notifications/compressImages
     grpcEndPoint: flair-engine:6565
-    FlairBiEndPoint: http://flair-bi:8002
     discovery:
         eureka:
             url: http://admin:admin@flair-registry:8761/eureka
@@ -19,12 +19,3 @@ data:
         caCert: certs/trustCertCollectionFile.crt
         serverCert: certs/certChainFile.crt
         serverKey: certs/privateKeyFile.key
-    mailService:
-        serive: sendgrid
-        sender: admin-vizcentric@gmail.com
-        testUser: admin@vizcentric.com
-        host: smtp.sendgrid.net
-        port: 587
-        auth:
-            user: <key name>
-            pass: <password>

--- a/flair-notifications/templates/deployment.yaml
+++ b/flair-notifications/templates/deployment.yaml
@@ -41,19 +41,15 @@ spec:
               containerPort: 8090
               protocol: TCP
         #   livenessProbe:
-        #     exec:
-        #       command:
-        #         - grpc_health_probe
-        #         - -addr=:8091
-        #         - -connect-timeout=3s
-        #     initialDelaySeconds: 125
+        #     httpGet:
+        #       path: /health
+        #       port: http
+        #     initialDelaySeconds: 180
         #     timeoutSeconds: 5
         #   readinessProbe:
-        #     exec:
-        #       command:
-        #         - grpc_health_probe
-        #         - -addr=:8091
-        #         - -connect-timeout=3s
+        #     httpGet:
+        #       path: /health
+        #       port: http
         #     initialDelaySeconds: 120
         #     timeoutSeconds: 5
           resources:
@@ -61,10 +57,20 @@ spec:
           volumeMounts:
             - name: configuration
               mountPath: /flair-config
+        {{- if .Values.secretName }}
+            - name: secrets
+              mountPath: "/opt/flair/secrets"
+              readOnly: true
+        {{- end }}
       volumes:
         - name: configuration
           configMap:
             name: {{ include "flair-notifications.fullname" . }}
+    {{- if .Values.secretName }}
+        - name: secrets
+          secret:
+            secretName: {{ .Values.secretName }}
+    {{- end }}
       {{- with .Values.nodeSelector }}
       nodeSelector:
         {{- toYaml . | nindent 8 }}

--- a/flair-notifications/templates/deployment.yaml
+++ b/flair-notifications/templates/deployment.yaml
@@ -51,18 +51,18 @@ spec:
             - name: http
               containerPort: 8090
               protocol: TCP
-        #   livenessProbe:
-        #     httpGet:
-        #       path: /health
-        #       port: http
-        #     initialDelaySeconds: 180
-        #     timeoutSeconds: 5
-        #   readinessProbe:
-        #     httpGet:
-        #       path: /health
-        #       port: http
-        #     initialDelaySeconds: 120
-        #     timeoutSeconds: 5
+          livenessProbe:
+            httpGet:
+              path: /health
+              port: http
+            initialDelaySeconds: 180
+            timeoutSeconds: 5
+          readinessProbe:
+            httpGet:
+              path: /health
+              port: http
+            initialDelaySeconds: 150
+            timeoutSeconds: 5
           resources:
             {{- toYaml .Values.resources | nindent 12 }}
           volumeMounts:

--- a/flair-notifications/templates/deployment.yaml
+++ b/flair-notifications/templates/deployment.yaml
@@ -26,11 +26,22 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-{{- if .Values.env }}
           env:
+            - name: FLAIR_COMPONENT
+              value: "flair-notifications"
+{{- if .Values.env }}
     {{- range $key, $val := .Values.env }}
             - name: {{ $key }}
               value: {{ $val | quote }}
+    {{- end}}
+{{- end }}
+{{- if .Values.secretEnv.keys }}
+    {{- range .Values.secretEnv.keys }}
+            - name: {{ . }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Values.secretEnv.name }}
+                  key: {{ . }}
     {{- end}}
 {{- end }}
           ports:
@@ -57,7 +68,7 @@ spec:
           volumeMounts:
             - name: configuration
               mountPath: /flair-config
-        {{- if .Values.secretName }}
+        {{- if .Values.secretMountName }}
             - name: secrets
               mountPath: "/opt/flair/secrets"
               readOnly: true
@@ -66,10 +77,10 @@ spec:
         - name: configuration
           configMap:
             name: {{ include "flair-notifications.fullname" . }}
-    {{- if .Values.secretName }}
+    {{- if .Values.secretMountName }}
         - name: secrets
           secret:
-            secretName: {{ .Values.secretName }}
+            secretName: {{ .Values.secretMountName }}
     {{- end }}
       {{- with .Values.nodeSelector }}
       nodeSelector:

--- a/flair-notifications/values.yaml
+++ b/flair-notifications/values.yaml
@@ -6,12 +6,14 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-notifications
-  tag: 1.3.12-SNAPSHOT.4
+  tag: 2.1.3-SNAPSHOT.119
   pullPolicy: IfNotPresent
 
 imagePullSecrets: []
 nameOverride: ""
 fullnameOverride: ""
+
+# secretName: flair-notifications-secret
 
 service:
   type: ClusterIP

--- a/flair-notifications/values.yaml
+++ b/flair-notifications/values.yaml
@@ -13,7 +13,8 @@ imagePullSecrets: []
 nameOverride: ""
 fullnameOverride: ""
 
-# secretName: flair-notifications-secret
+# Secret mount to be used if any secret files are needed to be added to the container during startup
+# secretMountName: flair-secret
 
 service:
   type: ClusterIP
@@ -25,6 +26,16 @@ env:
   SLEEP_TIME: 60
   EUREKA_URL: http://admin:admin@flair-registry:8761/eureka
   GRPC_SSL_ENABLED: false
+
+# Can be used if you want to set environment variables from a secret
+# The key names need to be the same as the environment variables you want to set
+# You would have to make sure the secret is created with the same key names as the environment variables you would like to set
+
+# secretEnv:
+#   name: flair-secret
+#   keys:
+#     - FLAIR_SECRET_USERNAME
+#     - FLAIR_SECRET_PASSWORD
 
 resources: {}
 

--- a/flair-notifications/values.yaml
+++ b/flair-notifications/values.yaml
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-notifications
-  tag: 2.1.3-SNAPSHOT.123
+  tag: 2.5.1
   pullPolicy: IfNotPresent
 
 imagePullSecrets: []
@@ -36,6 +36,13 @@ env:
 #   keys:
 #     - FLAIR_SECRET_USERNAME
 #     - FLAIR_SECRET_PASSWORD
+#     - DATABASE_USERNAME
+#     - DATABASE_PASSWORD
+#     - DATABASE_NAME
+#     - DATABASE_HOSTNAME
+#     - DATABASE_PORT
+#     - DATABASE_DIALECT
+#     - EUREKA_URL
 
 resources: {}
 

--- a/flair-notifications/values.yaml
+++ b/flair-notifications/values.yaml
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-notifications
-  tag: 2.1.3-SNAPSHOT.119
+  tag: 2.1.3-SNAPSHOT.123
   pullPolicy: IfNotPresent
 
 imagePullSecrets: []

--- a/flair-notifications/values.yaml
+++ b/flair-notifications/values.yaml
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-notifications
-  tag: 2.5.1
+  tag: 2.5.2
   pullPolicy: IfNotPresent
 
 imagePullSecrets: []

--- a/flair-registry/templates/deployment.yaml
+++ b/flair-registry/templates/deployment.yaml
@@ -25,11 +25,22 @@ spec:
         - name: {{ .Chart.Name }}
           image: "{{ .Values.image.repository }}:{{ .Values.image.tag }}"
           imagePullPolicy: {{ .Values.image.pullPolicy }}
-{{- if .Values.env }}
           env:
+            - name: FLAIR_COMPONENT
+              value: "flair-registry"
+{{- if .Values.env }}
     {{- range $key, $val := .Values.env }}
             - name: {{ $key }}
               value: {{ $val | quote }}
+    {{- end}}
+{{- end }}
+{{- if .Values.secretEnv.keys }}
+    {{- range .Values.secretEnv.keys }}
+            - name: {{ . }}
+              valueFrom:
+                secretKeyRef:
+                  name: {{ $.Values.secretEnv.name }}
+                  key: {{ . }}
     {{- end}}
 {{- end }}
           ports:
@@ -53,6 +64,12 @@ spec:
             timeoutSeconds: 5
           resources:
 {{ toYaml .Values.resources | indent 12 }}
+        {{- if .Values.secretMountName }}
+          volumeMounts:
+            - name: secrets
+              mountPath: "/opt/flair/secrets"
+              readOnly: true
+        {{- end }}
     {{- with .Values.nodeSelector }}
       nodeSelector:
 {{ toYaml . | indent 8 }}
@@ -64,4 +81,10 @@ spec:
     {{- with .Values.tolerations }}
       tolerations:
 {{ toYaml . | indent 8 }}
+    {{- end }}
+    {{- if .Values.secretMountName }}
+      volumes:
+        - name: secrets
+          secret:
+            secretName: {{ .Values.secretMountName }}
     {{- end }}

--- a/flair-registry/values.yaml
+++ b/flair-registry/values.yaml
@@ -6,7 +6,7 @@ replicaCount: 1
 
 image:
   repository: flairbi/flair-registry
-  tag: v5.0.3
+  tag: v5.0.8
   pullPolicy: Always
 
 nameOverride: ""

--- a/flair-registry/values.yaml
+++ b/flair-registry/values.yaml
@@ -12,6 +12,9 @@ image:
 nameOverride: ""
 fullnameOverride: ""
 
+# Secret mount to be used if any secret files are needed to be added to the container during startup
+# secretMountName: flair-secret
+
 service:
   type: ClusterIP
   eurekaPort: 8761
@@ -21,6 +24,16 @@ env:
   JHIPSTER_SLEEP: 0
   SPRING_PROFILES_ACTIVE: prod
   BITBUCKET_URI: https://github.com/viz-centric/flair-config.git
+
+# Can be used if you want to set environment variables from a secret
+# The key names need to be the same as the environment variables you want to set
+# You would have to make sure the secret is created with the same key names as the environment variables you would like to set
+
+# secretEnv:
+#   name: flair-secret
+#   keys:
+#     - FLAIR_SECRET_USERNAME
+#     - FLAIR_SECRET_PASSWORD
 
 resources: {}
 

--- a/secure-mesh.sh
+++ b/secure-mesh.sh
@@ -9,7 +9,7 @@ kubectl label namespace flair istio-injection=enabled
 docker pull flairbi/flair-registry:v5.0.6.ae1e037
 docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
-docker pull flairbi/flair-notifications:2.5.1
+docker pull flairbi/flair-notifications:2.5.2
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
 docker pull flairbi/flairbi:v2.5.1

--- a/secure-mesh.sh
+++ b/secure-mesh.sh
@@ -6,13 +6,13 @@ echo "[INFO] Creating Namespace 'Flair' with Istio Enabled"
 kubectl create ns flair
 kubectl label namespace flair istio-injection=enabled
 
-docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:v2.4.0
-docker pull flairbi/flair-cache:latest
-docker pull flairbi/flair-notifications:2.4.1
+docker pull flairbi/flair-registry:v5.0.8
+docker pull flairbi/flair-engine:v2.5.0
+docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
+docker pull flairbi/flair-notifications:2.5.1
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:v2.4.1-SNAPSHOT.eb12f144
+docker pull flairbi/flairbi:v2.5.1
 
 echo "[INFO] Installing Flair Registry"
 helm upgrade \

--- a/secure-mesh.sh
+++ b/secure-mesh.sh
@@ -6,7 +6,7 @@ echo "[INFO] Creating Namespace 'Flair' with Istio Enabled"
 kubectl create ns flair
 kubectl label namespace flair istio-injection=enabled
 
-docker pull flairbi/flair-registry:v5.0.8
+docker pull flairbi/flair-registry:v5.0.6.ae1e037
 docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
 docker pull flairbi/flair-notifications:2.5.1

--- a/secure-mesh.sh
+++ b/secure-mesh.sh
@@ -7,12 +7,12 @@ kubectl create ns flair
 kubectl label namespace flair istio-injection=enabled
 
 docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:latest
+docker pull flairbi/flair-engine:v2.4.0
 docker pull flairbi/flair-cache:latest
-docker pull flairbi/flair-notifications:1.3.12-SNAPSHOT.4
+docker pull flairbi/flair-notifications:2.4.1
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:latest
+docker pull flairbi/flairbi:v2.4.1-SNAPSHOT.eb12f144
 
 echo "[INFO] Installing Flair Registry"
 helm upgrade \

--- a/setup.sh
+++ b/setup.sh
@@ -6,7 +6,7 @@ kubectl create ns flair
 docker pull flairbi/flair-registry:v5.0.6.ae1e037
 docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
-docker pull flairbi/flair-notifications:2.5.1
+docker pull flairbi/flair-notifications:2.5.2
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
 docker pull flairbi/flairbi:v2.5.1

--- a/setup.sh
+++ b/setup.sh
@@ -4,12 +4,12 @@ echo "[INFO] Creating Namespace 'Flair'"
 kubectl create ns flair
 
 docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:v2.2.5-SNAPSHOT.c873a96
+docker pull flairbi/flair-engine:v2.4.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
-docker pull flairbi/flair-notifications:2.1.3-SNAPSHOT.119
+docker pull flairbi/flair-notifications:2.4.1
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:v2.2.1-SNAPSHOT.e1a39e43
+docker pull flairbi/flairbi:v2.4.1-SNAPSHOT.eb12f144
 
 echo "[INFO] Installing Flair Registry"
 helm upgrade \

--- a/setup.sh
+++ b/setup.sh
@@ -3,7 +3,7 @@
 echo "[INFO] Creating Namespace 'Flair'"
 kubectl create ns flair
 
-docker pull flairbi/flair-registry:v5.0.8
+docker pull flairbi/flair-registry:v5.0.6.ae1e037
 docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
 docker pull flairbi/flair-notifications:2.5.1

--- a/setup.sh
+++ b/setup.sh
@@ -4,12 +4,12 @@ echo "[INFO] Creating Namespace 'Flair'"
 kubectl create ns flair
 
 docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:latest
-docker pull flairbi/flair-cache:latest
-docker pull flairbi/flair-notifications:1.3.12-SNAPSHOT.4
+docker pull flairbi/flair-engine:v2.2.5-SNAPSHOT.c873a96
+docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
+docker pull flairbi/flair-notifications:2.1.3-SNAPSHOT.119
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:latest
+docker pull flairbi/flairbi:v2.2.1-SNAPSHOT.e1a39e43
 
 echo "[INFO] Installing Flair Registry"
 helm upgrade \

--- a/setup.sh
+++ b/setup.sh
@@ -3,13 +3,13 @@
 echo "[INFO] Creating Namespace 'Flair'"
 kubectl create ns flair
 
-docker pull flairbi/flair-registry:v5.0.3
-docker pull flairbi/flair-engine:v2.4.0
+docker pull flairbi/flair-registry:v5.0.8
+docker pull flairbi/flair-engine:v2.5.0
 docker pull flairbi/flair-cache:v2.2.2-SNAPSHOT.2f59747
-docker pull flairbi/flair-notifications:2.4.1
+docker pull flairbi/flair-notifications:2.5.1
 docker pull docker.io/bitnami/postgresql:10.7.0
 docker pull couchdb:2.3.1
-docker pull flairbi/flairbi:v2.4.1-SNAPSHOT.eb12f144
+docker pull flairbi/flairbi:v2.5.1
 
 echo "[INFO] Installing Flair Registry"
 helm upgrade \

